inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -0,0 +1,231 @@
+            dispatchData.lws = {1, 1, 1};
+        } else {
+            dispatchData.gws = {input.Batch().v, 1, 1};
+            dispatchData.lws = {1, 1, 1};
+        }
+    } else {
+        dispatchData.gws = {1, 1, 1};
+        dispatchData.lws = {1, 1, 1};
+    }
+
+    return dispatchData;
+}
+
+void DetectionOutputKernelRef::SetKernelArguments(const detection_output_params& params, clKernelData& kernel, size_t idx) const {
+    if (idx == 0) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 1});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 1) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 2) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    } else if (idx == 3) {
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1});
+        kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2});
+    }
+}
+
+KernelsData DetectionOutputKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    assert(params.GetType() == KernelType::DETECTION_OUTPUT && options.GetType() == KernelType::DETECTION_OUTPUT);
+
+    constexpr size_t kKernelsNum = 4;
+    KernelData kd = KernelData::Default<detection_output_params>(params, kKernelsNum);
+    const detection_output_params& detectOutParams = static_cast<const detection_output_params&>(params);
+
+    constexpr size_t prior_box_size = 4;
+    auto num_of_images = detectOutParams.inputs[0].Batch().v;
+    auto loc_feature_num = detectOutParams.inputs[0].Feature().v;
+    auto num_classes = detectOutParams.detectOutParams.num_classes;
+    auto num_loc_classes = (detectOutParams.detectOutParams.share_location) ? 1 : num_classes;
+    auto num_prior_boxes = (loc_feature_num / (num_loc_classes * prior_box_size));
+
+    constexpr size_t buffer_bytes = 16; // bboxes[xmin, ymin, xmax, ymax], scores[batchId, classId, boxId, score]
```

Comment:
it is preferable to calculate using size_of(dtype of each variable)

Reply:
Actually, because struct "Scores" is declared inside kernel(detection_output_kernel_ref.cl), So it cannot be referenced by sizeof(). Instead, I left a comment for this.